moto/sagemaker/models.py

```diff
@@ -353,6 +353,83 @@ def arn_for_model_name(model_name, region_name):
             + model_name
         )
 
+    @property
+    def physical_resource_id(self):
+        return self.model_arn
+
+    def get_cfn_attribute(self, attribute_name):
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-model.html#aws-resource-sagemaker-model-return-values
+        from moto.cloudformation.exceptions import UnformattedGetAttTemplateException
+
+        if attribute_name == "ModelName":
+            return self.model_name
+        raise UnformattedGetAttTemplateException()
+
+    @staticmethod
+    def cloudformation_name_type():
+        return None
+
+    @staticmethod
+    def cloudformation_type():
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-model.html
+        return "AWS::SageMaker::Model"
+
+    @classmethod
+    def create_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name
+    ):
+        sagemaker_backend = sagemaker_backends[region_name]
+
+        # Get required properties from provided CloudFormation template
+        properties = cloudformation_json["Properties"]
+        execution_role_arn = properties["ExecutionRoleArn"]
+        primary_container = properties["PrimaryContainer"]
+
+        # `create_model` returns `Model.response_create()` so we will
```

Comment:
It does at the moment.. but it probably shouldn't. Separation of concerns and all that.
Would you mind changing this so that the `sagemaker_backend.create_model` just returns a Model object, and the response.py than calls `response_create()`?
That way, the model is only concerned about storing the data - the response is responsible for formatting that data - and we don't have to manually create the model twice here